python/cuspatial/cuspatial/core/geodataframe.py

```diff
@@ -120,6 +126,35 @@ def _copy_type_metadata(
 
         return type_copied
 
+    def _slice(self: T, arg: slice) -> T:
+        """
+        Overload the _slice functionality from cudf's frame members.
+        """
+        # Collect the Geometry columns and slice them separately
+        columns_mask = cudf.Series(self.columns)
+        geocolumn_mask = pd.Series(
+            [isinstance(self[col], GeoSeries) for col in self.columns]
+        )
+        geo_names = columns_mask[geocolumn_mask].to_pandas()
+        geo_columns = self[columns_mask[geocolumn_mask].to_pandas()]
+        sliced_geo_columns = GeoDataFrame(
+            {name: geo_columns[name].iloc[arg] for name in geo_names}
+        )
+        # Send the rest of the columns to `cudf` to slice.
+        data_columns = cudf.DataFrame(
+            self[columns_mask[~geocolumn_mask].values_host]
+        )
+        sliced_data_columns = data_columns._slice(arg)
+        output = {
+            key: (
+                sliced_geo_columns[key] if value else sliced_data_columns[key]
+            )
+            for key, value in zip(
+                columns_mask.values_host, geocolumn_mask.values
+            )
```

Comment:
I'm a bit not sure if it's name should be called "key" and "value".
```suggestion
            name: (
                sliced_geo_columns[name] if mask else sliced_data_columns[name]
            )
            for name, mask in zip(
                columns_mask.values_host, geocolumn_mask.values
            )
```